src/Exceptions/BadRequest.php

```diff
@@ -23,9 +23,12 @@ public function __construct(ResponseInterface $response)
     {
         $this->response = $response;
 
-        $body = json_decode($response->getBody(), true);
+        $content = (string) $response->getBody();
+        $body = json_decode($content, true);
 
-        if ($body !== null) {
+        if (strlen($content) > 0 && is_null($body)) {
```

Comment:
It isn't readable what is happening here. Could you refactor this to a function with a good name? 

Also avoid using `else`.